subsys/bluetooth/host/adv.c

```diff
@@ -1469,6 +1485,17 @@ void bt_le_adv_resume(void)
 	if (IS_ENABLED(CONFIG_BT_PRIVACY) &&
 	    !atomic_test_bit(adv->flags, BT_ADV_USE_IDENTITY)) {
 		bt_id_set_adv_private_addr(adv);
+	} else {
+		uint8_t own_addr_type;
+		bool dir_adv = !bt_addr_le_eq(&adv->target_addr, BT_ADDR_LE_ANY);
+		uint32_t options = adv_get_options(adv);
+
+		/* Always set the address. Don't assume it has not changed. */
+		err = bt_id_set_adv_own_addr(adv, options, dir_adv, &own_addr_type);
```

Comment:
Seems like some refactoring/redesign might be in order here, so that we avoid doing the conversions between `options` and `flags` back and forth. You're already passing `adv` to `bt_id_set_adv_own_addr()`, so it shouldn't be necessary to pass the options separately to it (same information in two places).

Reply:
yep I thought the same, and actually started doing it. But then I figured it'd be better in a separate PR. The whole advertising module needs a big spring cleaning, that we're probably going to do this year.